src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,93 @@
+import React, { Children, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Tabs.scss';
+
+class Tabs extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        const { activeTabId } = props;
+
+        this.state = {
+            activeTabId,
+            derivedProp: activeTabId // eslint-disable-line react/no-unused-state
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.activeTabId === state.derivedProp) {
+            return null;
+        }
+        return {
+            activeTabId: props.activeTabId,
+            derivedProp: props.activeTabId
+        };
+    }
+
+    handleTabChange(e, id) {
+        e.preventDefault();
+        const { onChange } = this.props;
+        this.setState({
+            activeTabId: id
+        });
+        if (!onChange) {
+            return true;
+        }
+        return onChange(id);
+    }
+
+    render() {
+        const { activeTabId: idFromProps, children, onChange, ...rest } = this.props;
+        const { activeTabId } = this.state;
+
+        return (
+            <div {...rest} {...this.block()}>
+                <div {...this.elem('container')}>
+                    {Children.map(children, tab => {
+                        const { href, id } = tab.props;
+                        if (!id) {
+                            return null;
+                        }
+                        const active = activeTabId === id;
+                        return (
+                            <a
+                                href={href || '#'}
+                                {...this.elem(active ? 'tabActive' : 'tab')}
+                                key={id}
+                                onClick={e => {
```

Comment:
I've also checked the MaterialUI API for tabs. Seems like it really make sense to remove some control from Tabs component and give to to a user (developer) to make it more flexible.

As you can see below they only care about tabs navigation bar, and it's up to a user to render tab's content anywhere he/she wants. I can imagine it can be really useful in a real-world product. For sure it makes usage of the component bit trickier, but we can always create something like `SimpleTabs` component based on this one if we need it.

![image](https://user-images.githubusercontent.com/403250/50287266-6b005200-0462-11e9-8a4e-6c99b232e2d0.png)

Reply:
So if I get your point, you mean decoupling tabs and tab content but also adding one more wrapping component if you do want to use them next to each other (like it's now) Can you provide a use case / example for which this would be necessary (and cannot yet be done currently)? It feels to me like making it more flexible than we really need it to be..

Reply:
I just gave it a little try but it really is much more complicated and would need major overhauling. I think we need to discuss if this is something we really might need, and if so, we'll have to look at it together (it involves assigning matching ids between `Tab`'s and `TabContainer`'s, passing on callbacks back and forth, conditionally showing a certain `TabContainer` (it's not the only type of child within `SimpleTabs`) etc.